### ISSUE TITLE
[docdb] Investigate LogSegment and LogReader interaction when dealing with invalid segments

### ISSUE THREAD

**Issue by bmatican** [yugabyte/yugabyte-db#5920]
Noticed this during a crash loop on a TS. After fixing it, a bunch of wals were essentially empty, and caused various types of bootstrap failures:

```
	FAILED (TABLET_DATA_READY): Corruption (yb/consensus/log_reader.cc:207): Failed log replay. Reason: Failed to open new log: Segment sequence numbers are not consecutive. Previous segment: seqno 0, path /usr/scratch/yugabyte/data0/yb-data/tserver/wals/table-79ba82a966074cf9af535c8650f051ac/tablet-b2b7e962db58486cb65910aaf4891f0f/wal-000000054; Current segment: seqno 21, path /usr/scratch/yugabyte/data0/yb-data/tserver/wals/table-79ba82a966074cf9af535c8650f051ac/tablet-b2b7e962db58486cb65910aaf4891f0f/wal-000000021
```

During operations, one thing we tried was to just remove all of the wals, since we assumed they were all empty, but that lead to another failure, as there were no ops in the wals, and it failed a diff DB check.

This is likely related to #3783, as I noticed some potentially inconsistent pieces in the code, eg
```
287 Status ReadableLogSegment::ReadHeader() {
288   uint32_t header_size;
289   RETURN_NOT_OK(ReadHeaderMagicAndHeaderLength(&header_size));
290   if (header_size == 0) {
291     // If a log file has been pre-allocated but not initialized, then
292     // 'header_size' will be 0 even the file size is > 0; in this
293     // case, 'is_initialized_' remains set to false and return
294     // Status::OK() early. LogReader ignores segments where
295     // IsInitialized() returns false.
296     return Status::OK();
297   }
```
However this is at odds with the LogReader code
```
177     RETURN_NOT_OK_PREPEND(ReadableLogSegment::Open(env_, fqp, &segment),
178                           Format("Unable to open readable log segment: $0", fqp));
179     DCHECK(segment);
180     CHECK(segment->IsInitialized()) << "Uninitialized segment at: " << segment->path();
```
Moreover, if we were to legit skip these segments in the reader, we'd then fail slightly further down
```
204         return STATUS(Corruption, Substitute("Segment sequence numbers are not consecutive. "
205             "Previous segment: seqno $0, path $1; Current segment: seqno $2, path $3",
206             previous_seg_seqno, previous_seg_path,
207             entry->header().sequence_number(), entry->path()));
```
So it's likely that under some circumstances, we end up not registering these empty / uninitialized logs, while other times we register these just fine.
cc @hectorgcr @rahuldesirazu

**Comment by tedyu** [comment]
Previous segment: seqno 0
tablet-b2b7e962db58486cb65910aaf4891f0f/wal-000000054

Current segment: seqno 21
tablet-b2b7e962db58486cb65910aaf4891f0f/wal-000000021

I noticed that although seq no increased, the numerical values of suffix of wal files were decreasing.

**Comment by kmuthukk** [comment]
fix for this is under code review/testing.

**Comment by mbautin** [comment]
We probably need to backport the fix to 2.4, I think we see this in a customer cluster.

**Comment by mbautin** [comment]
A workaround for this issue is to delete the last WAL files that only contain zeroes. We need to create a tool for that.
